docs/practices/protocol_upgrade.md

```diff
@@ -23,6 +23,32 @@ especially if the change is invasive. For those changes, we may want to have
 several months of testing before we are confident that the change itself works
 and that it doesn't break other parts of the system.
 
+### Protocol version voting and upgrade
+
+When a new neard version, containing a new protocol version, is released, all node maintainers need 
+to upgrade their binary. That typically means stopping neard, downloading or compiling the new neard
+binary and restarting neard. However the protocol version of the whole network is not immediately 
+bumped to the new protocol version. Instead a process called voting takes place and determines if and 
+when the protocol version upgrade will take place. 
+
+Voting is a fully automated process in which all block producers across the network vote in support 
+or against upgrading the protocol version. The voting happens in the last block of an epoch after a 
```

Comment:
Voting happens at the end of every epoch.

The preconfigured voting date determines at which point the block producers stop pretending that they don't know anything about the new protocol version.